### PR TITLE
Fix warnings in akka-bench-jmh

### PR DESCRIPTION
**Pull request by helena**

### CODE REVIEW

File: akka-bench-jmh/src/main/scala/akka/actor/BenchmarkActors.scala [hunk 1]
```diff
@@ -93,12 +95,13 @@ object BenchmarkActors {
     def props(next: Option[ActorRef]) = Props(new Pipe(next))
   }
 
+  @silent("deprecated")
```
Comment (chbatey) [inline akka-bench-jmh/src/main/scala/akka/actor/BenchmarkActors.scala]: does this method need to use a deprecated api?

File: akka-bench-jmh/src/main/scala/akka/remote/artery/CodecBenchmark.scala [hunk 2]
```diff
@@ -38,6 +41,7 @@ import scala.concurrent.duration._
 class CodecBenchmark {
   import CodecBenchmark._
 
+  @silent("immutable val") // never updated, can not val due to: unbound placeholder parameter
```
Comment (chbatey) [inline akka-bench-jmh/src/main/scala/akka/remote/artery/CodecBenchmark.scala]: JMH updates this via reflection

File: akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala [hunk 3]
```diff
@@ -17,8 +19,10 @@ import org.reactivestreams.{ Publisher, Subscriber, Subscription }
 @State(Scope.Benchmark)
 @OutputTimeUnit(TimeUnit.MILLISECONDS)
 @BenchmarkMode(Array(Mode.Throughput))
+ */
+@silent // too long to add pattern
```
Comment (chbatey) [inline akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala]: maybe change the comment to give a clue as to what the warning is

File: akka-bench-jmh/src/main/scala/akka/actor/BenchmarkActors.scala (hunk 1 above)
Comment (helena) [reply akka-bench-jmh/src/main/scala/akka/actor/BenchmarkActors.scala]: Earlier deprecated was the pattern but on a retry with all the others in it does not which is mysterious.

File: akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala (hunk 3 above)
Comment (helena) [reply akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala]: It's a novel about a file with a package and nothing in it etc. Sure.

**Comment by helena** [comment]
@chbatey Suggestions pushed thanks

File: akka-bench-jmh/src/main/scala/akka/actor/BenchmarkActors.scala [hunk 4]
```diff
@@ -98,7 +98,7 @@ object BenchmarkActors {
     val fullPathToDispatcher = "akka.actor." + dispatcher
     val latch = new CountDownLatch(numPairs * 2)
     val actors = for {
-      i <- (1 to numPairs).toVector
+      _ <- (1 to numPairs).toVector
```
Comment (raboof) [inline akka-bench-jmh/src/main/scala/akka/actor/BenchmarkActors.scala]: Can this also be
```
List.fill(numPairs) {
  val ping = system.actorOf(PingPong.props(messagesPerPair, latch).withDispatcher(fullPathToDispatcher))
  val pong = system.actorOf(PingPong.props(messagesPerPair, latch).withDispatcher(fullPathToDispatcher))
  (ping, pong)
}
```
... but this is OK as well :D

File: akka-bench-jmh/src/main/scala/akka/serialization/jackson/JacksonSerializationBench.scala [hunk 5]
```diff
@@ -184,6 +185,7 @@ class JacksonSerializationBench {
   var system: ActorSystem = _
   var serialization: Serialization = _
 
+  @silent("immutable val") // never updated, can not val due to: unbound placeholder parameter
```
Comment (raboof) [inline akka-bench-jmh/src/main/scala/akka/serialization/jackson/JacksonSerializationBench.scala]:
```suggestion
  @silent("immutable val") // JMH updates this via reflection
```

File: akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala [hunk 6]
```diff
@@ -17,8 +19,13 @@ import org.reactivestreams.{ Publisher, Subscriber, Subscription }
 @State(Scope.Benchmark)
 @OutputTimeUnit(TimeUnit.MILLISECONDS)
 @BenchmarkMode(Array(Mode.Throughput))
+ */
+// Found names but no class, trait or object is defined in the compilation unit.
+// The incremental compiler cannot record the dependency information in such case.
+// Some errors like unused import referring to a non-existent class might not be reported.
```
Comment (raboof) [inline akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala]: Hrm this is pretty ugly... perhaps rename it to something other than `.scala` to sidestep this issue?

**Review by raboof — APPROVED** [review]

**Comment by raboof** [comment]
some suggestions/questions for consideration but generally LGTM

File: akka-bench-jmh/src/main/scala/akka/actor/BenchmarkActors.scala (hunk 4 above)
Comment (helena) [reply akka-bench-jmh/src/main/scala/akka/actor/BenchmarkActors.scala]: 👍 updating

File: akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala (hunk 6 above)
Comment (helena) [reply akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala]: Agree, Chris asked for noting more of the issue, so that's what it is :)

Comment (helena) [reply akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala]: @raboof `.bench`?

Comment (raboof) [reply akka-bench-jmh/src/main/scala/akka/stream/NewLayoutBenchmark.scala]: yeah or `.template` or `.txt`?

File: akka-bench-jmh/src/main/scala/akka/remote/artery/CodecBenchmark.scala (hunk 2 above)
Comment (helena) [reply akka-bench-jmh/src/main/scala/akka/remote/artery/CodecBenchmark.scala]: Updated

**Comment by helena** [comment]
@raboof All suggestions pushed thanks.

**Comment by helena** [comment]
Something off about this, post merge, so far sanity, closing and re-pushing with all changes made.

**Pull request closed without merging**